src/scenes/ProgressReports/Detail/ProgressReportDrawer/Steps/NextStepButton.tsx

```diff
@@ -0,0 +1,44 @@
+import { ArrowForward } from '@mui/icons-material';
+import { Box, Typography } from '@mui/material';
+import { StyleProps, Sx } from '~/common';
+import { useProgressReportContext } from '../../../ProgressReportContext';
+
+const typographyLinkSx: Sx = (theme) => ({
+  color: theme.palette.grey[800],
+  fontSize: 14,
+  cursor: 'pointer',
+  marginLeft: 2,
+  '&:hover': {
+    textDecoration: 'underline',
+  },
+});
+
+export const NextStepButton = ({ sx }: StyleProps) => {
+  const { nextProgressReportStep } = useProgressReportContext();
+
+  return (
+    <Box
```

Comment:
Can we just use an MUI `<Button>` here instead? This is a lot of work to make this half way between a link and a button. IMO it should just be a normal button style. If truly desired we could disableRipple and change hover to underline. It would just be less consistent with the rest of the app.

---

Also another pet peeve of mine is doing navigation in an `onClick`. Navigation should always be done with Link-type components (`Link`, `ButtonLink`, etc.) Those use anchor tags which allow "open in new tab/window" to work normally (via modifier key or from right click context menu).

The downside for this is that the `to` path needs to be determined ahead of time, which requires a slight refactor. The end result though is just swapping the `nextStep` function with a string.

Reply:
Let's hold off on Links, since that would move us away from `makeQueryHandler`. How we handle the query params could need to be customized at usage of the Link. Like do we merge or replace the params? Is it all or just our own? what are our own? etc.